streams/runners/storm/layout/src/main/java/com/hortonworks/streamline/streams/layout/storm/KafkaSourceScheme.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hortonworks.streamline.streams.layout.storm;
+
+import org.apache.storm.spout.MultiScheme;
+
+import java.util.Map;
+
+/**
+ * Scheme to be implemented for Kafka sources to deserialize bytebuffer record into tuples.
+ */
+public interface KafkaSourceScheme extends MultiScheme {
+
+    /**
+     * Key for topic property.
+     */
+    String TOPIC_KEY = "topic";
```

Comment:
>The constants can be moved to some other class like KafkaSpoutFluxComponent.

These constants are more specific to `KafkaSourceScheme` and I would like them to be declared here.

>We can accept MultiScheme implementations, instead of forcing them to implement KafkaSourceScheme always so that users can pass existing multi schemes. If the user has not provided any scheme, by default we can instantiate AvroKafkaSpoutScheme while building the flux component. If the users pass an instance of KafkaSourceScheme we invoke the init with the config.

Kafka source expects that user should register schema for a given topic and it does not allow configuring the source till the schema is available for the given topic. So, respective `MultiScheme` should be used for deserializing the messages. We can allow other MultiSchemes but those schemes may handle opaque data like string/binary without any structure defined by schema. I guess it is better to restrict users but there may be cases where producers may not be really using any schema registry and sends opaque data to respective topics. In that case it is good to allow user to configure respective MultiSchemes. I am fine in supporting this kind of usecase.

@harshach What is your opinion on this?